### ISSUE TITLE
./manage.py startpage [page]

### ISSUE THREAD

**Issue by onno-timmerman** [wagtail/wagtail#7873]
Wouldn't it be nice to have a command ./manage.py startpage in wagtail
it only should have the dir name [page] and __init__.py and models. The model can then be prefilled with a class and the some imports that are common.

just a suggestion
https://docs.wagtail.io/en/stable/reference/management_commands.html

**Comment by tomdyson** [comment]
e.g. `./manage.py startpage blog`

creates `./blog/models.py` with

```python
from wagtail.core.models import Page
from wagtail.core.fields import RichTextField
from wagtail.admin.edit_handlers import FieldPanel


class BlogPage(Page):
    body = RichTextField(blank=True)

    content_panels = Page.content_panels + [
        FieldPanel("body", classname="full"),
    ]
```

and `./blog/templates/blog/blog_page.py` with

```html
{% extends "base.html" %}
{% load wagtailcore_tags %}

{% block content %}
    {{ page.body|richtext }}
{% endblock %}
```

?

**Comment by onno-timmerman** [comment]
The template I would not do as mostly you create in in templates separate folder

**Comment by rinti** [comment]
We do something similar in Wagtail-Pipit.
`python manage.py new_page PageName`: https://github.com/Frojd/Wagtail-Pipit/blob/main/Company-Project/src/pipit/management/commands/new_page.py

**Comment by saevarom** [comment]
You would probably have to include the app name when you create the new model. This would be similar to other standard django commands where you have to supply the app name with the model name to namespace it:

`manage.py startpage blog.BlogPage`

**Comment by gasman** [comment]
I'm undecided on this... this would essentially be codifying the rule "every new page type belongs in its own app" as best practice, and I'm not sure if that's too opinionated for Wagtail itself. (It's certainly a better idea than putting everything in one enormous app, but I'd say it's equally valid to put a group of related page types in one app - e.g. a `blog` app containing `BlogPage`, `BlogIndex`, `BlogCategoryPage`.)

I'd be perfectly happy to see something like this in an add-on package though. `wagtail-torchbox-opinions`? :-)

If anything, I'd argue that the rule "you should keep your templates inside the app folder" is more standardised and less controversial, so if we were to go ahead with this, I'd be in favour of including the template too.

**Comment by onno-timmerman** [comment]
I don't see the problem, grouping models in one app is good practice. But maybe it should call then ./manage.py start_wag --models BlogPage BlogIndex BlogCategoryPage .

Maybe some name suggestions:

- ./ start_skeleton
- ./ start_section
- ./ start_cms

**Comment by rinti** [comment]
I agree with gasman that this is a pretty good case for a third party library.

**Comment by thibaudcolas** [comment]
I’m going to convert this to Discussions to make this a bit more visible. Cool idea, unsure whether it’s clear enough of an API to be implemented in core while leaving room for everyone’s opinions on how this should work.
